align/affineGap.go

```diff
@@ -1,259 +1,352 @@
 				mRowCurrent[2][j] = gapOpen
+				trace_prep_j[2][j/checkersize_j][i] = mRowCurrent[2][j]
 			} else if i == 0 {
 				mRowCurrent[0][j] = veryNegNum
 				mRowCurrent[1][j] = gapExtend + mRowCurrent[1][j-1]
-				trace[1][i][j] = ColI
 				mRowCurrent[2][j] = veryNegNum
+				if j%checkersize_j == 0 {
+					trace_prep_j[0][j/checkersize_j][i] = mRowCurrent[0][j]
+					trace_prep_j[1][j/checkersize_j][i] = mRowCurrent[1][j]
+					trace_prep_j[2][j/checkersize_j][i] = mRowCurrent[2][j]
+				}
 			} else if j == 0 {
 				mRowCurrent[0][j] = veryNegNum
 				mRowCurrent[1][j] = veryNegNum
 				mRowCurrent[2][j] = gapExtend + mRowPrevious[2][j]
-				trace[2][i][j] = ColD
+				trace_prep_j[0][j/checkersize_j][i] = mRowCurrent[0][j] //it is implied that j%checkersize_j==0. It must be saved in trace_prep_j
+				trace_prep_j[1][j/checkersize_j][i] = mRowCurrent[1][j]
+				trace_prep_j[2][j/checkersize_j][i] = mRowCurrent[2][j]
 			} else {
-				mRowCurrent[0][j], trace[0][i][j] = tripleMaxTrace(scoreColumnMatch(alpha, beta, i-1, j-1, scores)+mRowPrevious[0][j-1], scoreColumnMatch(alpha, beta, i-1, j-1, scores)+mRowPrevious[1][j-1], scoreColumnMatch(alpha, beta, i-1, j-1, scores)+mRowPrevious[2][j-1])
-				mRowCurrent[1][j], trace[1][i][j] = tripleMaxTrace(gapOpen+gapExtend+mRowCurrent[0][j-1], gapExtend+mRowCurrent[1][j-1], gapOpen+gapExtend+mRowCurrent[2][j-1])
-				mRowCurrent[2][j], trace[2][i][j] = tripleMaxTrace(gapOpen+gapExtend+mRowPrevious[0][j], gapOpen+gapExtend+mRowPrevious[1][j], gapExtend+mRowPrevious[2][j])
+				mRowCurrent[0][j], _ = tripleMaxTrace(scores[alpha[i-1]][beta[j-1]]+mRowPrevious[0][j-1], scores[alpha[i-1]][beta[j-1]]+mRowPrevious[1][j-1], scores[alpha[i-1]][beta[j-1]]+mRowPrevious[2][j-1])
+				mRowCurrent[1][j], _ = tripleMaxTrace(gapOpen+gapExtend+mRowCurrent[0][j-1], gapExtend+mRowCurrent[1][j-1], gapOpen+gapExtend+mRowCurrent[2][j-1])
+				mRowCurrent[2][j], _ = tripleMaxTrace(gapOpen+gapExtend+mRowPrevious[0][j], gapOpen+gapExtend+mRowPrevious[1][j], gapExtend+mRowPrevious[2][j])
+				if j%checkersize_j == 0 {
+					trace_prep_j[0][j/checkersize_j][i] = mRowCurrent[0][j]
+					trace_prep_j[1][j/checkersize_j][i] = mRowCurrent[1][j]
+					trace_prep_j[2][j/checkersize_j][i] = mRowCurrent[2][j]
+				}
 			}
 		}
-		if i < mColumn-1 {
+
+		if i%checkersize_i == 0 && i < mColumn-1 {
+			copy(trace_prep_i[0][i/checkersize_i], mRowCurrent[0]) //copy instead of assign values to avoid pointers automatically updating/synchronizing the 2 matrices
+			copy(trace_prep_i[1][i/checkersize_i], mRowCurrent[1])
+			copy(trace_prep_i[2][i/checkersize_i], mRowCurrent[2])
 			mRowPrevious, mRowCurrent = mRowCurrent, mRowPrevious
+		} else if i < mColumn-1 {
+			mRowPrevious, mRowCurrent = mRowCurrent, mRowPrevious //reuse mRow variables to save memory, but only up until the second to last row
 		}
 	}
-	maxScore, route := affineTrace(mRowCurrent, mColumn, trace)
-
-	return maxScore, route
+	score_highest_i := mColumn - 1
```

Comment:
Right now it looks like we use the final corner spot in the matrix as the "highest score."  This would be a "global alignment" that forces all of sequence one to align to all of sequence two.  In the future, we can add an option for a "loca alignment" where would would actually keep track of the highest score in the matrix and send back that score and its coordinates in the matrix (not needed for this pull request).